### ISSUE TITLE
Phase 1.3: CodeTrek Module

### ISSUE THREAD

**Issue by GauravGusain98** [ColoredCow/portal#2794]
**Feature Suggested By**
@Ayush8923 

**Parent Issue:-** https://github.com/ColoredCow/portal/issues/2704

**Related Issue**
https://github.com/ColoredCow/portal/issues/2734
https://github.com/ColoredCow/portal/issues/2736
https://github.com/ColoredCow/portal/issues/2743

**Is your feature request related to a problem? Please describe.**
In ColoredCow, CodeTrek is a journey that most of the interns do. In it, they learn the basics of software development and hone their skills. Now, we want to track and capture this journey of all the interns in Portal. This will help us to monitor the progress of each individual and see where we can or they can do better. 

For this, we have introduced Codetrek Module and created a form to add applicants in previous the [task](https://github.com/ColoredCow/portal/issues/2743). Now we will create DB schema to store details corresponding to applicant

**Point of contact for guidance**
@GauravGusain98

**Comment by GauravGusain98** [comment]
### Action Items
**Basics - 1 pointer (Skip if you've already done these in your previous tasks)**
- [ ] Read about Models in Laravel. Share your understanding of why we use them in the task comment in your own language
- [ ] Read about Views in Laravel. Share your understanding of why we use them in the task comment in your own language
- [ ] Read about Routes in Laravel. Share your understanding of why we use them in the task comment in your own language
- [ ] Read about Controller in Laravel. Share your understanding of why we use them in the task comment in your own language
- [ ] Read about Services files in Laravel. Share your understanding of why we use them in the task comment in your own language
- [ ] Read how we can print a value in Laravel using dd()
- [ ] Try printing `hello world` using the `dd` function

**Database Migrations - 1 pointer**
- [ ] Read about [Laravel Migration](https://laravel.com/docs/8.x/migrations#creating-tables). Write your understanding in the comments. Also mention the use of `up` and `down` function in migration file.
- [ ] Create a table `codetrek_applicants` in the employee portal database using this [Schema](https://docs.google.com/presentation/d/1WC_lr2P1C6O6L6pobD2sxtL-A3YIfDr8AYOeU1x9CVQ/edit#slide=id.g1bc21c5bde4_0_5)

**Form Submission - 1 pointer**
- [ ] Read about how forms are submitted in Laravel. You can take reference from [here](https://vegibit.com/how-to-set-up-form-submission-in-laravel/)
- [ ] Now make a function in Controller to store the applicant data 
- [ ] Once your form is ready for submission, try printing the data that you are sending from the form using the dd () function
- [ ] Read about different types of [resource functions](https://laravel.com/docs/8.x/controllers#resource-controllers) in laravel Controller and their use. Write your understanding in the comments

**Eloquent - 1 pointer**
- [ ] Read how we can create an entry in a table using the Laravel model [create function](https://laravel.com/docs/8.x/eloquent-relationships#the-create-method)
- [ ] Store the applicant data in the codetrek_applicants table

**Frontend - 1 pointer** 
- [ ] Make a service file, named as `CodeTrekService` and write a function to fetch all applicant's details.
- [ ] Send the details to CodeTrek page
- [ ] On CodeTrek page show the list of applicants under the applicant tab 

**Test the functionality**
- [ ] Try to find the case where the created functionality can break. If any found fix them
- [ ] Assign PR for review

**Comment by nishant22029** [comment]
Laravel Migration
•It is a powerful way to organize and manage database changes. 
•Migrations are like version control for your database,it helps to easily modify and share the application's database schema.
•Laravel Migration allows us to add a new column or delete the records in your database without deleting 
 the records that are already present.

 use of up and down function in migration file
•In migration files, the up and down functions are used to create and remove tables, columns, indexes, and constraints.
The up function is used to add new tables, columns, indexes, and constraints to the database, while the down function is
used to remove them.

Different types of resource functions in laravel controller
•index() Function: This function is used to list all the resources from a model. It is usually used to render a view to 
 display a collection of resources.

•store() Function: This function is used to create a new resource in the database. It takes the data from a form and saves it to the model.

•show() Function: This function is used to display a single resource. It takes the provided id and retrieves the resource from the database.

•update() Function: This function is used to update an existing resource in the database. It takes the data from a form and updates the resource
 with the new values.

•destroy() Function: This function is used to delete a resource from the database. It takes the provided id and removes the resource from the database.
